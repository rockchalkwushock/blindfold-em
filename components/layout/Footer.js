import { styles } from '../../lib'

/**
 * @function Footer
 * @description renders styled <footer />
 *
 * @prop {Array} children
 * @returns React Element
 */
const Footer = ({ children }) => (
  <footer>
    {children}
    <style jsx>{`
      footer {
        background-color: ${styles.colors.lowLight};
        display: flex;
        flex-direction: column;
        font-size: 1.2em;
        height: 10vh;
        justify-content: center;
        padding: 1em;
      }
      @media (orientation: landscape) {
        footer {
          font-size: 1em;
        }
      }
    `}</style>
  </footer>
)

export default Footer
