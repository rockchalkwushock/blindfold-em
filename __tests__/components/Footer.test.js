import React from 'react'
import { mount } from 'enzyme'

import Footer from '../../components/layout/Footer'

const Test = () => (
  <Footer>
    <div>1</div>
    <div>2</div>
    <div>3</div>
  </Footer>
)

describe('Component: <Footer />', () => {
  test('should mount without exploding', () => {
    const tree = mount(<Test />)
    expect(tree).toMatchSnapshot()
  })
})
