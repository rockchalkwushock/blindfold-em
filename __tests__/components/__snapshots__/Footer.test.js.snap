// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Footer /> should mount without exploding 1`] = `
<Test>
  <Footer>
    <footer
      className="jsx-1106431596"
    >
      <div>
        1
      </div>
      <div>
        2
      </div>
      <div>
        3
      </div>
      <JSXStyle
        css="footer.jsx-1106431596{background-color:lightcoral;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;height:15vh;padding:1em;}"
        styleId="1106431596"
      />
    </footer>
  </Footer>
</Test>
`;
